snuba/reader.py

```diff
@@ -31,6 +28,36 @@
 else:
     Result = MutableMapping[str, Any]
 
+
+def iterate_rows(result: Result) -> Iterator[Row]:
+    if "totals" in result:
+        return itertools.chain(result["data"], [result["totals"]])
+    else:
+        return iter(result["data"])
+
+
+def build_result_transformer(
+    column_transformations: Mapping[str, Callable[[Any], Any]],
```

Comment:
This approach also should map more cleanly to streaming results, since this only requires knowledge of the schema upfront and then each row can be transformed independently. (I didn't update this implementation to transform on a per-row basis instead of a per-result basis to minimize the complexity here, but it would be a pretty straightforward change.)